interactors/singleTransactionAddressNonceHandler_test.go

```diff
@@ -0,0 +1,238 @@
+package interactors
+
+import (
+	"context"
+	"crypto/rand"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/testsCommon"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSingleTransactionAddressNonceHandler_NewSingleTransactionAddressNonceHandler(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil proxy", func(t *testing.T) {
+		t.Parallel()
+
+		anh, err := NewSingleTransactionAddressNonceHandler(nil, nil)
+		assert.Nil(t, anh)
+		assert.Equal(t, ErrNilProxy, err)
+	})
+	t.Run("nil addressHandler", func(t *testing.T) {
+		t.Parallel()
+
+		anh, err := NewSingleTransactionAddressNonceHandler(&testsCommon.ProxyStub{}, nil)
+		assert.Nil(t, anh)
+		assert.Equal(t, ErrNilAddress, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		pubkey := make([]byte, 32)
+		_, _ = rand.Read(pubkey)
+		addressHandler := data.NewAddressFromBytes(pubkey)
+
+		_, err := NewSingleTransactionAddressNonceHandler(&testsCommon.ProxyStub{}, addressHandler)
+		assert.Nil(t, err)
+	})
+}
+
+func TestSingleTransactionAddressNonceHandler_ApplyNonce(t *testing.T) {
+	t.Parallel()
+
+	t.Run("proxy returns error should error", func(t *testing.T) {
+
+		proxy := &testsCommon.ProxyStub{
+			GetAccountCalled: func(address core.AddressHandler) (*data.Account, error) {
+				return nil, expectedErr
+			},
+		}
+
+		anh, _ := NewSingleTransactionAddressNonceHandler(proxy, testAddress)
+
+		txArgs := createTxArgs()
+
+		err := anh.ApplyNonce(context.Background(), &txArgs)
+		require.Equal(t, expectedErr, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+
+		blockchainNonce := uint64(100)
+		proxy := &testsCommon.ProxyStub{
+			GetAccountCalled: func(address core.AddressHandler) (*data.Account, error) {
+				return &data.Account{Nonce: blockchainNonce}, nil
+			},
+		}
+
+		anh, _ := NewSingleTransactionAddressNonceHandler(proxy, testAddress)
+
+		txArgs := createTxArgs()
+
+		err := anh.ApplyNonce(context.Background(), &txArgs)
+		require.Nil(t, err)
+		require.Equal(t, blockchainNonce, txArgs.Nonce)
+	})
+}
+
+func TestSingleTransactionAddressNonceHandler_fetchGasPriceIfRequired(t *testing.T) {
+	t.Parallel()
```

Comment:
can be moved inside t.Run

Reply:
I am not sure about that. Moving it inside t.Run wouldn\`t make `TestSingleTransactionAddressNonceHandler_fetchGasPriceIfRequired` be executed only after the other `Test...` function will be executed, and not in parallel ?

Reply:
as discussed, we`ll make the t.Run just a simple comment